tests/route/test_route_flap.py

```diff
@@ -61,12 +74,26 @@ def get_ptf_recv_ports(duthost, tbinfo):
 
 
 def get_ptf_send_ports(duthost, tbinfo, dev_port):
-    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
-    member_port = mg_facts['minigraph_portchannels'][dev_port]['members']
-    send_port = mg_facts['minigraph_ptf_indices'][member_port[0]]
+    if tbinfo['topo']['name'] in ['t0', 't1-lag']:
+        mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+        member_port = mg_facts['minigraph_portchannels'][dev_port]['members']
+        send_port = mg_facts['minigraph_ptf_indices'][member_port[0]]
+    else:
+        mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+        ports = natsorted(mg_facts['minigraph_ports'].keys())
+        send_port = mg_facts['minigraph_ptf_indices'][ports[0]]
     return send_port
 
 
+def check_route(duthost, route, dev_port, operation):
+    out = json.loads(duthost.shell('vtysh -c "show ip route {} json"'.format(route), verbose=False)['stdout'])
+    result = [nexthop['interfaceName'] for nexthop in out[route][0]['nexthops']]
+    if operation == WITHDRAW:
+        pytest_assert(dev_port not in result, "Route {} was not withdraw {}".format(route, result))
+    else:
+        pytest_assert(dev_port in result, "Route {} was not announced {}".format(route, result))
+
```

Comment:
This may take some time from ASIC sync with sonic, I think using `wait_until` is better here ?

Reply:
@yutongzhang-microsoft, 
I used sleep for 1 second before calling this functions, and it looked like it was enough.